### PR TITLE
Add Nabla axes for Extrusion Depth and Edge Highlight

### LINKED ISSUE #58: Add Highlight (HLIT) axis
@justvanrossum required this for Nabla

* https://github.com/justvanrossum/nabla/issues/22

### PR DESCRIPTION
**Pull request by davelab6**
Ref #58 and #60. Don't close those until axis images and description review is completed.

cc @justvanrossum

### CODE REVIEW

**Comment by davelab6** [comment]
This needs images before merging.

File: Lib/axisregistry/data/extrusion_depth.textproto
Comment (justvanrossum) [inline Lib/axisregistry/data/extrusion_depth.textproto:9]:
```suggestion
  "Controls extrusion depth."
```

Comment (chrissimpkins) [inline Lib/axisregistry/data/extrusion_depth.textproto:9]:
```suggestion
  "Controls extrusion depth."
```

**Review by chrissimpkins — CHANGES_REQUESTED** [review]

**Review by justvanrossum — COMMENTED** [review]
How mandatory is that default value? I hope not at all...

File: Lib/axisregistry/data/extrusion_depth.textproto
Comment (justvanrossum) [reply Lib/axisregistry/data/extrusion_depth.textproto:9]: Chris beat me to it :)

Comment (chrissimpkins) [reply Lib/axisregistry/data/extrusion_depth.textproto]: added in https://github.com/googlefonts/axisregistry/pull/61/commits/bd67f5b0a8c46a31ab14fcc608c63689334a6741

**Comment by davelab6** [comment]
> How mandatory is that default value? I hope not at all...

Correct, not at all, we have per-family override ability, but the system requires a library-wide default

**Comment by rsheeter** [comment]
> How mandatory is that default value?

It doesn't have to be your axis default in the font, it's just what you get from our API by default. If that's a bad default for your family it can be overriden per family. 

That said ... @davelab6 having the only family that implements an axis override the default makes me wonder if it might be the wrong default? Maybe by default we should have a little extrusion and highlighting?

**Comment by davelab6** [comment]
Since it always need to be overridden the default is meaningless.

**Comment by rsheeter** [comment]
> please suggest a value

If you have these axes I would think the default should show them off a little, such as on fonts.google home or specimen. On that basis my naive thought would be we pick some non-zero value that makes sense for Nabla, ideally one that we think might possibly also make sense if someone else implements these axes.

@justvanrossum what default would you choose on that basis?

**Comment by davelab6** [comment]
In `shear_and_extrude.py` the nabla defaults are 50% of the range implemented

https://github.com/justvanrossum/nabla/blob/a1a463e3c8c285675331dca3d7f794a1ea23f75e/scripts/shear_and_extrude.py#L463-L487

**Comment by justvanrossum** [comment]
> the default should show them off a little, such as on fonts.google home or specimen

Yes, the default from the font is perfect for that. A non-font-specific default doesn't make sense.

**Comment by davelab6** [comment]
I've updated the PR with the Nabla values - @rsheeter does this now look good?

**Comment by rsheeter** [comment]
LGTM. Is the image/description review a blocker for merging? - would love to have fully functional Nabla on dev.

**Comment by chrissimpkins** [comment]
> LGTM. Is the image/description review a blocker for merging? - would love to have fully functional Nabla on dev.

We'll need to add a single fallback to each new registration configuration (like https://github.com/googlefonts/axisregistry/pull/66/files) to land this without the images and description review

**Comment by justvanrossum** [comment]
If there's anything specific that Arthur or I can do, please let us know.

FWIF, I  feel it is counter-productive to *require* axis registration in the first place. It makes sense for axes that are (or will be) common, but for all we know, Nabla might be the only font to ever need a "Highlight" axis. Any new font may introduce an axis that hasn't been thought of before, and the mandatory GF axis registration is just burocracy that doesn't really benefit anyone. It's holding back innovation.

**Comment by justvanrossum** [comment]
Also, the fact that an axis registration contains a mandatory default value is misguided: it is the font that should decide what the axis default is. It makes some sense for "wght" and "wdth" to have a generally accepted default value, for other axes not so much.

**Comment by rsheeter** [comment]
>  it is the font that should decide what the axis default is

The axis registry default doesn't have to match the font default for that axis, it's merely what our API gives you if you didn't specify anything for that axis and didn't override for the family.  

For context, we use the axis registry to give our systems some metadata about axes. It's also, probably it's primary purpose, intended to help us avoid having multiple definitions of the same axis. For example, four fonts on Google Fonts that all define grade with a wildly different range wouldn't be a great outcome as it moves away from having standard well understood axes. On that note, I hope in time to see some axes emerge as semi-standard and to have widely accepted ranges and defaults as weight or width do. Grade comes to mind.

**Pull request merged**
